core/gaspool.go

```diff
@@ -25,6 +25,15 @@ import (
 // in a block. The zero value is a pool with zero gas available.
 type GasPool uint64
 
+// SetGas set an initial value for gaspool
+func (gp *GasPool) SetGas(amount uint64) *GasPool {
```

Comment:
This function seems useless, we can use `new(GasPool).AddGas(gasLimit)` to replace.

Reply:
If use `AddGas` instead, may lead to one of these results: 
- reallocate a new `big.Int` struct after each transaction be executed
- set a magic number(much bigger than `block.gaslimit`)
- do some extra calculation to increase up the gas been consumed.